### DISCUSSION TITLE
Moving EB support

### DISCUSSION THREAD

**Discussion by colinmcnally** [AMReX-Codes/amrex#2914] [Q&A] [answered]
I'm trying to trace down the code behind a recent AMReX based paper:

A moving embedded boundary approach for the compressible Navier-Stokes equations in a block-structured adaptive refinement framework
Natarajan, Mahesh; Grout, Ray; Zhang, Weiqun; Day, Marc
https://ui.adsabs.harvard.edu/abs/2022JCoPh.46511315N/abstract

I've found the docs on EB
https://amrex-codes.github.io/amrex/docs_html/EB_Chapter.html#

and I've noticed that AMReX-Hydro refers to a suspiciously named macro AMREX_USE_MOVING_EB,  but I've not succeeded in finding what I recognize as moving EB support infrastructure in the AMReX repo. Am I looking in the right place?

**Comment by drummerdoc** [comment]
Hi, the code used for that paper is in a private repo and is not part of the AMReX standard distribution.  If you want to explore that work further, we can take the query offline and I can help you do that.  Send me a note at Marc.Day@nrel.gov and I'm be happy to help.

**Comment by nataraj2** [comment, marked as answer]
Hi Colin,

I will be happy to help with the moving boundary code.

--
Mahesh Natarajan
________________________________

**Reply by itai001** [reply]
Hi nataraj2,
When I try to compile the examples in MovingEB_CNSAMReX, I get compilation errors. Could you please give me some suggestions on how to solve them?

/home/hongchang/lianzifan/MovingEB_CNSAMReX/CNS_EBoft_Expansion/Source/main.cpp: In function ‘int main(int, char**)’:
/home/hongchang/lianzifan/MovingEB_CNSAMReX/CNS_EBoft_Expansion/Source/main.cpp:53:6: error: no matching function for call to ‘amrex::Amr::Amr()’
   53 |  Amr amr;
      |      ^~~
In file included from /home/hongchang/lianzifan/MovingEB_CNSAMReX/CNS_EBoft_Expansion/Source/main.cpp:5:
/home/hongchang/lianzifan/amrex/Src/Amr/AMReX_Amr.H:42:5: note: candidate: ‘amrex::Amr::Amr(const amrex::RealBox*, int, const amrex::Vector<int>&, int, amrex::LevelBld*)’
   42 |     Amr (const RealBox* rb, int max_level_in, const Vector<int>& n_cell_in, int coord,
      |     ^~~
/home/hongchang/lianzifan/amrex/Src/Amr/AMReX_Amr.H:42:5: note:   candidate expects 5 arguments, 0 provided
/home/hongchang/lianzifan/amrex/Src/Amr/AMReX_Amr.H:40:5: note: candidate: ‘amrex::Amr::Amr(amrex::LevelBld*)’
   40 |     Amr (LevelBld* a_levelbld /* One must pass LevelBld* as an argument now*/);
      |     ^~~
/home/hongchang/lianzifan/amrex/Src/Amr/AMReX_Amr.H:40:5: note:   candidate expects 1 argument, 0 provided
make: *** [tmp_build_dir/o/3d.gnu.MPI.EXE/main.o] Error 1
make: *** Waiting for unfinished jobs....

**Reply by WeiqunZhang** [reply]
You need to make the following change due to changes in amrex.
```
-        Amr amr;
+        Amr amr(getLevelBld());
```
You might also need to declare the function, if the compiler complains about it. Just put this line aboove `int main (...)`.
```
amrex::LevelBld* getLevelBld ();
```
